spec/requests/admin_claims_spec.rb

```diff
@@ -71,6 +71,37 @@
             expect(response.body).to include(claim_with_matching_attributes.reference)
           end
         end
+
+        context "when the claim is amendable" do
+          let(:claim) { create(:claim, :submitted) }
+
+          it "displays a link to amend the claim" do
+            get admin_claim_path(claim)
+
+            expect(response.body).to include("Amend claim")
+          end
+        end
+
+        context "when the claim is not amendable" do
+          let(:claim) { create(:claim, :rejected) }
+
+          it "does not display a link to amend the claim" do
+            get admin_claim_path(claim)
+
+            expect(response.body).not_to include("Amend claim")
+          end
+        end
+
+        context "when the claim has been amended" do
+          let(:claim) { create(:claim, :submitted, teacher_reference_number: "1234567") }
+          let!(:amendment) { create(:amendment, claim: claim, claim_changes: {teacher_reference_number: ["7654321", "1234567"]}) }
+
+          it "renders a timeline of changes" do
+            get admin_claim_path(claim)
+
+            expect(response.body).to include("changed from 7654321 to 1234567")
```

Comment:
Request for opinions: I'm a bit uneasy about the lack of request specs for the claim timeline, but it's hard to test without parsing the HTML in these specs, because the text is split over various HTML tags. It's just about possible to make a useful-ish test for the case where personal data hasn't been removed, as I have here, but it's impossible for the case where personal data has been removed, where the only string I can search for is "changed"...

So the timeline is mainly tested via tests of the `admin/timeline_helper` spec, and I'm wondering whether that's enough.

Reply:
Could you use a feature test and check for the presence / absence of particular selectors on a page? Something like:

```ruby
expect(page).to have_selector("p.hmcts-timeline__by[text='changed']")
```

Reply:
Yeah, I've added a feature spec for it. Didn't actually need to go to the level of checking for individual tags; the `have_content` matcher does a pretty good job of smooshing everything together so I can just do

```ruby
expect(page).to have_content("Teacher reference number\nchanged from 7654321 to 1234567")
expect(page).to have_content("Bank account number\nchanged")
expect(page).not_to have_content("Bank account number\nchanged from")
```

Reply:
Sweet 👍